docs/adr/0002-shopping-lists-lineitem-and-textlineitem-update-actions.md

```diff
@@ -0,0 +1,332 @@
+                with <b>SKU-1</b>, so we need to find another way to create a new line item with the right order.</p>
+        </td>
+    </tr>
+    <tr>
+        <th colspan="2">Proposed solution</th>
+    </tr>
+    <tr>
+        <td colspan="2">
+            <p>
+                Normally, for a difference, we might do a set intersection and then calculate action for differences, 
+                but that does not make sense because we are not aware of the order from the draft. 
+                So in this case the one request could be created but we might need to remove line item with <b>SKU-2</b>
+                and line items in the draft with the given order with the line items <b>SKU-1</b> and <b>SKU-2</b>.
+            </p>
+            <pre lang="json">
+{
+  "version": 1,
+  "actions": [
+    {
+      "action": "removeLineItem",
+      "lineItemId": "24de3821-e27d-4ddb-bd0b-ecc99365285f"
+    },
+    {
+      "action": "addLineItem",
+      "sku": "SKU-1",
+      "quantity": 1,
+      "addedAt": "2020-11-04T09:38:35.571Z",
+      "custom": {
+        "type": {
+          "key": "custom-type-for-shoppinglists"
+        },
+        "fields": {
+          "textField": "text-1"
+        }
+      }
+    },
+    {
+      "action": "addLineItem",
+      "sku": "SKU-2",
+      "quantity": 2,
+      "addedAt": "2020-11-04T09:40:12.341Z",
+      "custom": {
+        "type": {
+          "key": "custom-type-for-shoppinglists"
+        },
+        "fields": {
+          "textField": "text-2"
+        }
+      }
+    }
+  ]
+}
+</pre>
+        </td>
+    </tr>
+</table>
+
+### Do we need to remove all line items when the order changes ?
+
+<table>
+    <tr>
+        <th>LineItemDrafts</th>
+        <th>LineItems</th>
+    </tr>
+    <tr>
+        <td><pre lang="json">
+{
+  "lineItems": [
+    {
+      "sku": "SKU-1",
+      "quantity": 1,
+      "addedAt": "2020-11-04T09:38:35.571Z",
+      "custom": {
+        "type": {
+          "key": "custom-type-for-shoppinglists"
+        },
+        "fields": {
+          "textField": "text-1"
+        }
+      }
+    },
+    {
+      "sku": "SKU-3",
+      "quantity": 3,
+      "addedAt": "2020-11-05T10:00:10.101Z",
+      "custom": {
+        "type": {
+          "key": "custom-type-for-shoppinglists"
+        },
+        "fields": {
+          "textField": "text-3"
+        }
+      }
+    }
+  ]
+}
+</pre>
+        </td>
+        <td><pre lang="json">
+{
+  "lineItems": [
+    {
+      "id": "24de3821-e27d-4ddb-bd0b-ecc99365285f",
+      "variant": {
+        "sku": "SKU-1"
+      },
+      "quantity": 2,
+      "custom": {
+        "type": {
+          "id": "4796e155-f5a4-403a-ae1a-04b10c9dfc54",
+          "obj": {
+            "key": "custom-type-for-shoppinglists"
+          }
+        },
+        "fields": {
+          "textField": "text-1"
+        }
+      },
+     "addedAt": "2020-11-04T09:38:35.571Z"
+    },
+    {
+      "id": "24de3821-e27d-4ddb-bd0b-ecc99365285f",
```

Comment:
Is it supposed to have different ID as LineItem SKU-1?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/614/commits/8fe412126d47738a3bbd4c44933bedbca7e3133c